packages/flame/lib/src/cache/images.dart

```diff
@@ -45,6 +45,19 @@ class Images {
     _assets[name] = _ImageAsset.fromImage(image);
   }
 
+  /// If the image with [name] exists in the cache that is returned, otherwise
+  /// the image generated by [imageGenerator] is returned.
+  ///
+  /// If the [imageGenerator] is used, the resulting [Image] is stored with
+  /// [name] in the cache.
+  Future<Image> fetchOrGenerate(
```

Comment:
Is this meant to fetch and if it fails to do so then, it builds an image and also caches it?

Reply:
Yes, is that not clear from the Dartdocs? 😅